### PR TITLE
Change repo name to mattermost-plugin-mscalendar

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR renames the repo from `*msoffice` to `*mscalendar`, as the plugin is specific to Microsoft Calendar part of the Microsoft Office suite.

Various display name changes in:
- plugin.json
- README.md
- server/config/const.go

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21531

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,14 +32,12 @@ follow [these instructions](#2-configuration) for install and configuration.
...
-  - Redirect URI -> `<MM_SITEURL>/plugins/com.mattermost.msoffice/oauth2/complete`
+  - Redirect URI -> `<MM_SITEURL>/plugins/com.mattermost.mscalendar/oauth2/complete`
     - For development (use ngrok.io URL)
 
 ### Step 2 Configure Plugin Settings
 
-**`System Console` > `PLUGINS` > `MS Office Calendar`**
-
-- [ ] (TODO: rename in plugin settings - currently `TODO:name`)
+**`System Console` > `PLUGINS` > `Microsoft Calendar`**
```
Comment (mickmister) [inline README.md]: README display name change

File: plugin.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-    "id": "com.mattermost.msoffice",
-    "name": "TODO:name",
-    "description": "TODO: description.",
+    "id": "com.mattermost.mscalendar",
+    "name": "Microsoft Calendar",
+    "description": "Microsoft Calendar Integration",
```
Comment (mickmister) [inline plugin.json]: plugin.json display name changes

File: server/config/const.go [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
+	BotUserName    = "mscalendar"
+	BotDisplayName = "Microsoft Calendar Bot"
+	BotDescription = "Created by the Microsoft Calendar Plugin."
 
-	ApplicationName = "Microsoft Office"
-	Repository      = "mattermost-plugin-msoffice"
-	CommandTrigger  = "msoffice"
+	ApplicationName = "Microsoft Calendar"
+	Repository      = "mattermost-plugin-mscalendar"
+	CommandTrigger  = "mscalendar"
```
Comment (mickmister) [inline server/config/const.go]: bot and command display name changes

Comment (levb) [reply server/config/const.go]: Suggest `mscal` for command? @aaronrothschild

Comment (levb) [reply server/config/const.go]: (can be a separate change though, not blocking for this PR)

**Pull request merged**